### PR TITLE
Masking passwords for alarmcallbacks in web interface.

### PR DESCRIPTION
**Pull request by dennisoelkers**
Adding functionality in rest client which masks password fields of configurable antity.

Usage should be extended to other entities like Outputs which have dynamic configurations too, but this is beyond the scope of this PR.

### CODE REVIEW

File: graylog2-rest-client/src/main/java/org/graylog2/restclient/models/AlarmCallback.java [hunk 1]
```diff
@@ -66,10 +67,15 @@ public String getType() {
         return type;
     }
 
+    @Override
     public Map<String, Object> getConfiguration() {
         return configuration;
     }
 
+    public Map<String, Object> getConfiguration(GetSingleAvailableAlarmCallbackResponse typeSummaryResponse) {
```
Comment (bernd) [inline graylog2-rest-client/src/main/java/org/graylog2/restclient/models/AlarmCallback.java]: Maybe change the parameter name to better match the type here?

File: graylog2-rest-client/src/main/java/org/graylog2/restclient/models/ConfigurableEntity.java [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+import org.graylog2.restclient.lib.plugin.configuration.RequestedConfigurationField;
+import org.graylog2.restclient.models.api.responses.alarmcallbacks.GetSingleAvailableAlarmCallbackResponse;
+
+import java.util.List;
+import java.util.Map;
+
+public abstract class ConfigurableEntity {
+    public abstract Map<String, Object> getConfiguration();
+
+    public Map<String, Object> getConfiguration(List<RequestedConfigurationField> typeResponses) {
```
Comment (bernd) [inline graylog2-rest-client/src/main/java/org/graylog2/restclient/models/ConfigurableEntity.java]: Maybe change the parameter name to better match the type here?

File: graylog2-rest-client/src/main/java/org/graylog2/restclient/models/Input.java [hunk 3]
```diff
@@ -264,18 +264,17 @@ private Long getGaugeValue(String name) {
     public Map<String, Object> getAttributes() {
         return attributes;
     }
-    public Map<String, Object> getAttributes(InputTypeSummaryResponse typeSummaryResponse ) {
-        Map<String, Object> result = Maps.newHashMapWithExpectedSize(attributes.size());
+    public Map<String, Object> getAttributes(List<RequestedConfigurationField> typeSummaryResponse ) {
```
Comment (bernd) [inline graylog2-rest-client/src/main/java/org/graylog2/restclient/models/Input.java]: Maybe change the parameter name to better match the type here?

**Comment by kroepke** [comment]
verified working. editing alarmcallbacks isn't supported yet, so that part naturally wasn't tested.

**Pull request merged**